src/modules/tas_studio/mod.rs

```diff
@@ -1560,7 +1638,82 @@ pub unsafe fn maybe_receive_messages_from_remote_server(marker: MainThreadMarker
             editor.recompute_extra_camera_frame_data_if_needed();
         }
         State::PreparingToPlayToEditor(_, _, _) => unreachable!(),
+        _ => (),
+    }
+}
+
+pub fn should_draw_pause(marker: MainThreadMarker) -> bool {
+    !matches!(*STATE.borrow_mut(marker), State::PlayingViews { .. })
+}
+
+pub fn tas_playback_rendered_views(marker: MainThreadMarker) {
+    if !TasStudio.is_enabled(marker) {
+        return;
     }
+
+    let mut state = STATE.borrow_mut(marker);
+    *state = match mem::take(&mut *state) {
+        State::PlayingViews {
+            editor,
+            last_generation,
+            last_branch_idx,
+            simulate_at,
+            bridge,
+            current_frame,
+        } => {
+            if current_frame == editor.stop_frame() as usize
+                || current_frame >= editor.branch().frames.len()
+            {
+                sdl::set_relative_mouse_mode(marker, false);
+                client::activate_mouse(marker, false);
+                engine::prepend_command(marker, "bxt_hud 1; hud_draw 1\n");
+                ENABLE_FREECAM_ON_CALCREFDEF.set(marker, true);
+
+                State::Editing {
+                    editor,
+                    last_generation,
+                    last_branch_idx,
+                    simulate_at,
+                    bridge,
+                }
+            } else {
+                let r_refdef_vieworg = unsafe { &mut *engine::r_refdef_vieworg.get(marker) };
+                let r_refdef_viewangles = unsafe { &mut *engine::r_refdef_viewangles.get(marker) };
+
+                r_refdef_vieworg[0] = editor.branch().frames[current_frame].state.player.pos[0];
+                r_refdef_vieworg[1] = editor.branch().frames[current_frame].state.player.pos[1];
+                r_refdef_vieworg[2] = editor.branch().frames[current_frame].state.player.pos[2];
+
+                // We don't keep track of vieworg so just deduce it from here instead.
+                // It is not easy to keep track of vieworg because it isn't there to track.
+                // vieworg is calculated.
+                // It is also funny how vieworg is calculated here,
+                // but viewangles is passed from BXT.
```

Comment:
I was gonna ask why this is necessary but I guess bxt-strafe just doesn't handle angle overrides yet?

Ideally it would handle overrides (then they will also work without simulation), but I guess that's effort?

Reply:
Yea I'd say bxt-strafe is not aware of the viewangles / rendered angle data so you might call this precursor to that. This is just half of the work if the simulation wants the overrides. However, this one is specifically just the rendered yaw, not the actual player yaw.